test-examples.sh

```
@@ -0,0 +1,350 @@
+    local example_dir="examples/$example_name"
+    shift
+    local expected_files=("$@")
+    
+    echo -e "${YELLOW}Testing $example_name...${NC}"
+    
+    # Check if example directory exists
+    if [ ! -d "$example_dir" ]; then
+        echo -e "${RED}✗ Example directory $example_dir does not exist${NC}"
+        echo -e "${RED}   Current directory: $(pwd)${NC}"
+        echo -e "${RED}   Available examples:${NC}"
+        ls -la examples/ 2>/dev/null | grep '^d' | awk '{print $NF}' | grep -v '^\.$\|^\.\.$' | sed 's/^/     - /'
```

Shellcheck (0.10.0)

[warning] Don't use ls | grep. Use a glob or a for loop with a condition to allow non-alphanumeric filenames.

(SC2010)